sdk/core/src/CommonNetworkMember/BaseNetworkMember.ts

```diff
@@ -0,0 +1,1151 @@
+    const {
+      issuerUrl,
+      revocationUrl,
+      metricsUrl,
+      registryUrl,
+      verifierUrl,
+      bloomVaultUrl,
+      affinidiVaultUrl,
+    } = basicOptions
+
+    const keysService = new KeysService(encryptedSeed, password)
+    this._metricsService = new MetricsService({
+      metricsUrl,
+      accessApiKey: accessApiKey,
+      component: component,
+    })
+
+    const sdkVersion = extractSDKVersion()
+    this._registryApiService = new RegistryApiService({ registryUrl, accessApiKey, sdkVersion })
+    this._issuerApiService = new IssuerApiService({ issuerUrl, accessApiKey, sdkVersion })
+    this._verifierApiService = new VerifierApiService({ verifierUrl, accessApiKey, sdkVersion })
+    this._revocationApiService = new RevocationApiService({ revocationUrl, accessApiKey, sdkVersion })
+    this._keyManagementService = createKeyManagementService(options)
+    this._didDocumentService = new DidDocumentService(keysService)
+    const didAuthService = new DidAuthService({ encryptedSeed, encryptionKey: password })
+    this._walletStorageService = new WalletStorageService(didAuthService, keysService, platformEncryptionTools, {
+      bloomVaultUrl,
+      affinidiVaultUrl,
+      accessApiKey,
+      storageRegion,
+      audienceDid: this.did,
+    })
+    this._jwtService = new JwtService()
+    this._holderService = new HolderService({ registryUrl, metricsUrl, accessApiKey }, component)
+    this._affinity = new (AffinidiCommon ?? (Affinity as AffinidiCommonConstructor))({
+      apiKey: accessApiKey,
+      registryUrl: registryUrl,
+      metricsUrl: metricsUrl,
+      component: component,
+    })
+    this._keysService = keysService
+
+    this._options = options
+    this._component = component
+    this._encryptedSeed = encryptedSeed
+    this._password = password
+    this._did = null
+    this._didDocumentKeyId = null
+    this._platformEncryptionTools = platformEncryptionTools
+  }
+
+  /**
+   * @description Returns user's encrypted seed
+   * @returns encrypted seed
+   */
+  get encryptedSeed(): string {
+    return this._encryptedSeed
+  }
+
+  /**
+   * @description Returns user's password
+   * @returns encrypted seed
+   */
+  get password(): string {
+    return this._password
+  }
+
+  /**
+   * @description Creates DID and anchors it
+   * 1. generate seed/keys
+   * 2. build DID document
+   * 3. sign DID document
+   * 4. store DID document in IPFS
+   * 5. anchor DID with DID document ID from IPFS
+   * @param password - encryption key which will be used to encrypt randomly created seed/keys pair
+   * @param options - optional parameter { registryUrl: 'https://affinity-registry.dev.affinity-project.org' }
+   * @returns
+   *
+   * did - hash from public key (your decentralized ID)
+   *
+   * encryptedSeed - seed is encrypted by provided password. Seed - it's a source to derive your keys
+   */
+  protected static async _register(
```

Comment:
@inga-affinidi @DenisPopov15 should we make register as a public method? We had a chat about it few days ago

Reply:
Isn't the point of these changes move user/login/auth related stuff out of the base network member and related core classes? Maybe this function can be taken out to another class instead of making it available publicly in the BaseNetworkMember?

Reply:
@Volland @Yengas , the public method is `register` in `LegacyNetworkMember` and `createWallet` in `NetworkMemberWithoutCognito`